File: 프로그래머스/lv0/120875. 평행/평행.java
class Solution {
    public int solution(int[][] dots) {
        for (int i = 1; i < dots.length; i++) {
            double d1 = (double) (dots[0][1] - dots[i][1]) / (dots[0][0] - dots[i][0]);
            double d2 = (double) (dots[i % 3 + 1][1] - dots[(i + 1) % 3 + 1][1]) / (dots[i % 3 + 1][0] - dots[(i + 1) % 3 + 1][0]);

            if (d1 == d2) return 1;
        }
        
        return 0;
    }
}

File: 프로그래머스/lv0/120875. 평행/평행.py
def solution(dots):
    answer=0
    for i in range(1, len(dots)):
        a=(dots[0][0]-dots[i][0])/(dots[0][1]-dots[i][1])
        b=(dots[(i%3)+1][0]-dots[((i+1)%3)+1][0])/(dots[(i%3)+1][1]-dots[((i+1)%3)+1][1])
        if a==b:
                answer=1
    return answer